cri-o/cri-o
# Close temporary image in PullImage

Signed-off-by: Marc Burns <m4burns@uwaterloo.ca>



**- What I did**

The `ImageServer` interface in `pkg/storage` documents that images returned by `PrepareImage()` should have `Close()` called. `PullImage()` currently fails to close an image returned by `PrepareImage()`, which causes `/var/tmp` to fill with large temporary files when using at least the `docker-archive` transport. I fixed this issue.

**- How I did it**

I modified the `ImageServer` interface to allow `Close()` to be called on images returned by `PrepareImage()`. I added a deferred call to `Close()` in `PullImage()`.

**- How to verify it**

I updated the unit tests to check that `Close()` is being called as prescribed by the `ImageServer` interface.

**- Description for the changelog**

Fix temporary file leak when using `docker-archive` transport.
